internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -495,10 +495,18 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 		// this prevents multiple XRs composing the same resource to be
 		// continuously alternated as controllers.
 		if err := c.client.Patch(ctx, cd.Resource, client.Apply, client.ForceOwnership, client.FieldOwner(ComposedFieldOwnerName(xr))); err != nil {
+			if kerrors.IsInvalid(err) {
+				// We don't want to return an error here, because we want to
+				// continue to apply other composed resources. We'll just return
+				// an event and mark the resource as not ready
+				events = append(events, event.Warning(reasonCompose, errors.Wrapf(err, errFmtApplyCD, name)))
```

Comment:
Currently when using functions `events` is only used to accumulate "results" that are returned by functions. This information is useful but I'm not sure how I feel about mixing in events produced by the `Compose` code as well.

Reply:
True, but I don't see any issue extending its usage to events caused by other non-terminal issues.